tests/system_tests/psyclone/test_psyclone_system_test.py

```diff
@@ -170,42 +176,44 @@ def test_run(self, config, psyclone_lfric_api):
         # So use a list instead:
         assert all(list(config.prebuild_folder.glob(f)) == [] for f in expect_prebuild_files)
         assert all(list(config.build_output.glob(f)) == [] for f in expect_build_files)
-        with config, pytest.warns(UserWarning, match="no transformation script specified"):
-            self.steps(config, psyclone_lfric_api)
+        with warns(UserWarning, match="no transformation script specified"):
+            self.steps(config)
         assert all(list(config.prebuild_folder.glob(f)) != [] for f in expect_prebuild_files)
         assert all(list(config.build_output.glob(f)) != [] for f in expect_build_files)
 
-    def test_prebuild(self, tmp_path, config, psyclone_lfric_api):
-        with config, pytest.warns(UserWarning, match="no transformation script specified"):
-            self.steps(config, psyclone_lfric_api)
+    def test_prebuild(self, config: BuildConfig) -> None:
+        """
+        Tests prebuilds are not rebuild the second time round.
+        """
+        config.prebuild_folder.mkdir(parents=True)
+
+        with warns(UserWarning, match="no transformation script specified"):
+            self.steps(config)
+        first_timestamps = {file: file.stat() for file in config.prebuild_folder.iterdir()}
 
-        # make sure no work gets done the second time round
-        with mock.patch('fab.parse.x90.X90Analyser.walk_nodes') as mock_x90_walk, \
-                mock.patch('fab.parse.fortran.FortranAnalyser.walk_nodes') as mock_fortran_walk, \
-                mock.patch('fab.tools.psyclone.Psyclone.process') as mock_run, \
-                config, pytest.warns(UserWarning, match="no transformation script specified"):
-            self.steps(config, psyclone_lfric_api)
+        with warns(UserWarning, match="no transformation script specified"):
+            self.steps(config)
+        second_timestamps = {file: file.stat() for file in config.prebuild_folder.iterdir()}
 
-        mock_x90_walk.assert_not_called()
-        mock_fortran_walk.assert_not_called()
-        mock_run.assert_not_called()
+        assert second_timestamps == first_timestamps
 
 
 class TestTransformationScript:
     """
     Check whether transformation script is called with x90 file once
     and whether transformation script is passed to psyclone after '-s'.
 
+    ToDo: Mockkery not appropriate in system tests.
     """
-    def test_transformation_script(self, psyclone_lfric_api):
+    def test_transformation_script(self) -> None:
         psyclone_tool = Psyclone()
         psyclone_tool._version = (2, 4, 0)
         psyclone_tool._is_available = True
 
         mock_transformation_script = mock.Mock(return_value=__file__)
         with mock.patch('fab.tools.psyclone.Psyclone.run') as mock_run_command:
             mock_transformation_script.return_value = Path(__file__)
-            psyclone_tool.process(api=psyclone_lfric_api,
+            psyclone_tool.process(api='dynamo0.3',
```

Comment:
While I am ok with removing the `psyclone_lfric_api` definintion (which allowed them all to be changed at one location), you then need to switch to `lfric`, not `dynamo0.3`, which is deprecated.